Sources/Commander/ConfigCommands.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  ConfigCommands.swift
+//  swiftgen
+//
+//  Created by Olivier Halligon on 01/10/2017.
+//  Copyright © 2017 AliSoftware. All rights reserved.
+//
+
+import Commander
+import PathKit
+import StencilSwiftKit
+import SwiftGenKit
+
+extension Config.Entry {
+  func checkPaths() throws {
+    for inputPath in self.paths {
+      guard inputPath.exists else {
+        throw Config.Error.pathNotFound(path: inputPath)
+      }
+    }
+    guard self.output.parent().exists else {
+      throw Config.Error.pathNotFound(path: self.output.parent())
+    }
+  }
+
+  func run(parserCommand: ParserCommand) throws {
+    let parser = try parserCommand.parserType.init(options: [:], warningHandler: { (msg, _, _) in
+      logMessage(.error, msg)
+    })
+    try parser.parse(paths: self.paths)
+    do {
+      let templateRealPath = try self.template.resolvePath(forSubcommand: parserCommand.name)
+      let template = try StencilSwiftTemplate(templateString: templateRealPath.read(),
+                                              environment: stencilSwiftEnvironment())
+
+      let context = parser.stencilContext()
+      let enriched = try StencilContext.enrich(context: context, parameters: self.parameters)
+      let rendered = try template.render(enriched)
+      let output = OutputDestination.file(self.output)
+      output.write(content: rendered, onlyIfChanged: true)
+    } catch let error as TemplateRef.Error {
+      logMessage(.error, error)
+    } catch let error {
+      logMessage(.error, "failed to render template: \(error)")
+    }
+  }
+}
+
+// MARK: - Commands
+
+// MARK: Lint
+
+let configLintCommand = command(
+  Option<Path>("config", default: "swiftgen.yml", flag: "c",
+               description: "Path to the configuration file to use",
+               validator: checkPath(type: "config file") { $0.isFile })
+) { file in
+  logMessage(.info, "Linting \(file)")
+  let config = try Config(file: file)
+  config.lint { level, msg in logMessage(level, msg) }
+}
+
+// MARK: Run
+
+let configRunCommand = command(
+  Option<Path>("config", default: "swiftgen.yml", flag: "c",
+               description: "Path to the configuration file to use",
+               validator: checkPath(type: "config file") { $0.isFile }),
+  Flag("verbose", default: false, flag: "v",
+       description: "Print each command being executed")
+) { file, verbose in
+  let config = try Config(file: file)
+
+  if verbose {
```

Comment:
Not really needed right now, but wouldn't it be nicer to have a logger system that accepts a logging level (info or above, warning or above, etc...) and the verbose flag modifies this (global) logger level? See comment below in the actual logger.

Reply:
Yeah, thought the same too. For now I didn't want to refactor _everything_, that PR was already the occasion of a little refactoring of other stuff (reorg files, finally making TemplateRef an enum as it was always meant to be exclusive choice, making it possible to provide the logging method for `Config.lint` so I can write proper tests on it to check if we logged the proper stuff…)

But for now I think it's not the most urgent thing to do. That reworking of `printError(string:)` to the new simplistic but level-based logging function was just a first step. Providing the log level on the CLI means that all commands should support it, meaning that we might need to tweak with Commander and the ArgumentParser even further, so that's be for later in a less urgent PR 😜